src/sentry/static/sentry/app/components/events/interfaces/frame/frameDefaultTitle.tsx

```diff
@@ -0,0 +1,175 @@
+
+    return {
+      key: 'module',
+      value: frame.module,
+      meta: getMeta(frame, 'module'),
+    };
+  };
+
+  // TODO(dcramer): this needs to use a formatted string so it can be
+  // localized correctly
+  if (defined(frame.filename || frame.module)) {
+    // prioritize module name for Java as filename is often only basename
+    const shouldPrioritizeModuleName =
+      framePlatform === 'java' || framePlatform === 'csharp';
+
+    const pathName = getPathName(shouldPrioritizeModuleName);
+    const enablePathTooltip = defined(frame.absPath) && frame.absPath !== pathName.value;
+
+    title.push(
+      <Tooltip key={pathName.key} title={frame.absPath} disabled={!enablePathTooltip}>
+        <code key="filename" className="filename">
+          {pathName.meta ? (
+            <AnnotatedText
+              value={<Truncate value={pathName.value} maxLength={100} leftTrim />}
+              chunks={pathName.meta.chunks}
+              remarks={pathName.meta.rem}
+              errors={pathName.meta.err}
+            />
+          ) : (
+            <Truncate value={pathName.value} maxLength={100} leftTrim />
+          )}
+        </code>
+      </Tooltip>
+    );
+
+    // in case we prioritized the module name but we also have a filename info
+    // we want to show a litle (?) icon that on hover shows the actual filename
+    if (shouldPrioritizeModuleName && frame.filename && framePlatform !== 'csharp') {
+      title.push(
+        <Tooltip key={frame.filename} title={frame.filename}>
+          <a className="in-at real-filename">
+            <IconQuestion size="xs" />
+          </a>
+        </Tooltip>
+      );
+    }
+
+    if (frame.absPath && isUrl(frame.absPath)) {
+      title.push(
+        <ExternalLink
+          href={frame.absPath}
+          className="icon-open"
+          key="share"
+          onClick={handleExternalLink}
+        />
+      );
+    }
+
+    if (defined(frame.function) || defined(frame.rawFunction)) {
+      title.push(
+        <span className="in-at" key="in">
+          {` ${t('in')} `}
+        </span>
+      );
+    }
+  }
+
+  if (defined(frame.function) || defined(frame.rawFunction)) {
+    title.push(<FrameFunctionName frame={frame} key="function" className="function" />);
+  }
+
+  // we don't want to render out zero line numbers which are used to
+  // indicate lack of source information for native setups.  We could
+  // TODO(mitsuhiko): only do this for events from native platforms?
+  if (defined(frame.lineNo) && frame.lineNo !== 0) {
+    title.push(
+      <span className="in-at in-at-line" key="no">
+        {` ${t('at line')} `}
+      </span>
+    );
+    title.push(
+      <code key="line" className="lineno">
+        {defined(frame.colNo) ? `${frame.lineNo}:${frame.colNo}` : frame.lineNo}
+      </code>
+    );
+  }
+
+  if (defined(frame.package) && framePlatform !== 'csharp') {
+    title.push(
+      <span className="within" key="within">
+        {` ${t('within')} `}
```

Comment:
This is a better solution than the `{' '}` approach.

Reply:
do you mean the `'  ' + t('within') + '  ' `approach? I still would prefer to use margin here, but it something I decided to introduce afterwards

Reply:
I think he means something like `{' '}{t('within')}{' '}`

Reply:
I was trying to say that I like the new implementation you have more than the previous one.

Reply:
yeah this I got 😊 glad you like it!